webui/react/src/hooks/useAuthCheck.ts

```diff
@@ -42,18 +43,24 @@ const useAuthCheck = (): (() => Promise<boolean>) => {
     const authToken = jwtToken ?? cookieToken ?? globalStorage.authToken;
 
     /*
-     * If auth token found, update the API bearer token and validate it with the current user API.
+     * If an auth token is found, validate it with the current user API, then update
+     * the API bearer token and set the user as authenticated.
      * If an external login URL is provided, redirect there.
      * Otherwise mark that we checked the auth and skip auth token validation.
      */
 
     if (authToken) {
-      updateBearerToken(authToken);
-
-      Observable.batch(() => {
-        authStore.setAuth({ isAuthenticated: true, token: authToken });
-        authStore.setAuthChecked();
-      });
+      try {
+        await getCurrentUser({});
+        updateBearerToken(authToken);
+        Observable.batch(() => {
+          authStore.setAuth({ isAuthenticated: true, token: authToken });
+        });
+      } catch (e) {
+        // If an invalid auth token is detected we need to properl handle the auth error
```

Comment:
**nit:** typo `properly`

Reply:
**question:** if it fails, should the source of the token get cleared? Things like:
- `jwt` token from searchParams
- `cookieToken` not sure if anything can or needs to be done here
- `globalStorage.authToken` clear this?

Reply:
Calling `handleError` ended up being the best approach here. I ran into unwanted beahvior/bugs otherwise. I am happy to delve into it though if need be. 

The `handleError` method eventually calls `paths.logout` which properly entirely resets the application and removes the tokens.

Regarding removing the `jwt` from the search params. I am not sure that we want to remove that here, since we have logic that explicitly checks for it and I am not sure if removing it will cause unwanted side effects. But if we think it is okay to remove here then I am open to it.